ts/twitter/src/app/home/_components/timeline/timeline-feed-service.ts

```diff
@@ -0,0 +1,80 @@
+import { ERROR_MESSAGES } from "@/lib/constants/error-messages";
+import { Post } from "@/lib/models/post";
+import { FakeTimelineFeedService } from "./__fake__/fake-timeline-feed-service";
+
+export type TimelineEventResponse =
+  | TimelineAccessedResponse
+  | PostCreatedResponse
+  | PostDeletedResponse;
+
+export type TimelineEventType = "TimelineAccessed" | "PostCreated" | "PostDeleted";
+
+interface TimelineAccessedResponse {
+  event_type: TimelineEventType & "TimelineAccessed";
+  posts: Post[];
+}
+
+interface PostCreatedResponse {
+  event_type: TimelineEventType & "PostCreated";
+  posts: Post[];
+}
+
+interface PostDeletedResponse {
+  event_type: TimelineEventType & "PostDeleted";
+  posts: Post[];
+}
+
+export interface TimelineFeedService {
+  connect(
+    url: string,
+    handleResponse: (response: TimelineEventResponse) => void,
+    handleError: (message: string) => void
+  ): void;
+  disconnect(): void;
+}
+
+export class SseTimelineFeedServicel implements TimelineFeedService {
+  private eventSource: EventSource | null = null;
+
+  connect(
+    url: string,
+    handleResponse: (response: TimelineEventResponse) => void,
+    handleError: (message: string) => void
+  ): void {
+    if (this.eventSource) return;
+
+    const eventSource = new EventSource(url);
+    this.eventSource = eventSource;
+
+    eventSource.onmessage = (event) => {
+      try {
+        const response: TimelineEventResponse = JSON.parse(event.data);
+        handleResponse(response);
+      } catch (err) {
+        handleError(ERROR_MESSAGES.INVALID_DATA);
+      }
+    };
+
+    eventSource.onerror = () => {
+      handleError(ERROR_MESSAGES.SERVER_ERROR);
+      this.disconnect();
+    };
+  }
+
+  disconnect(): void {
+    this.eventSource?.close();
+    this.eventSource = null;
+  }
+}
+
+let testInstance: FakeTimelineFeedService | null = null;
+
+export const CreateTimelineFeedService = (): TimelineFeedService => {
```

Comment:
```suggestion
export const createTimelineFeedService = (): TimelineFeedService => {
```
FYI: https://google.github.io/styleguide/tsguide.html#naming-rules-by-identifier-type